gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegistrationUnit.java

```diff
@@ -0,0 +1,222 @@
+  protected final State props = new State();
+  protected final State storageProps = new State();
+  protected final State serDeProps = new State();
+
+  /**
+   * Table or Partition properties
+   */
+  protected Optional<Long> createTime;
+  protected Optional<Long> lastAccessTime;
+
+  /**
+   * Storage properties
+   */
+  protected Optional<String> location;
+  protected Optional<String> inputFormat;
+  protected Optional<String> outputFormat;
+  protected Optional<Boolean> isCompressed;
+  protected Optional<Integer> numBuckets;
+  protected Optional<List<String>> bucketColumns;
+  protected Optional<Boolean> isSotredAsSubDirs;
+
+  /**
+   * SerDe properties
+   */
+  protected Optional<String> serDeType;
+
+  HiveRegistrationUnit(Builder<?> builder) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(builder.dbName));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(builder.tableName));
+
+    this.dbName = builder.dbName;
+    this.tableName = builder.tableName;
+    this.columns.addAll(builder.columns);
+    this.props.addAll(builder.props);
+    this.storageProps.addAll(builder.storageProps);
+    this.serDeProps.addAll(builder.serDeProps);
+
+    this.createTime = this.props.contains(HiveConstants.CREATE_TIME)
+        ? Optional.of(this.props.getPropAsLong(HiveConstants.CREATE_TIME)) : Optional.<Long> absent();
+    this.lastAccessTime = this.props.contains(HiveConstants.LAST_ACCESS_TIME)
+        ? Optional.of(this.props.getPropAsLong(HiveConstants.LAST_ACCESS_TIME)) : Optional.<Long> absent();
+
+    this.location = Optional.fromNullable(this.storageProps.getProp(HiveConstants.LOCATION));
+    this.inputFormat = Optional.fromNullable(this.storageProps.getProp(HiveConstants.INPUT_FORMAT));
+    this.outputFormat = Optional.fromNullable(this.storageProps.getProp(HiveConstants.OUTPUT_FORMAT));
+    this.isCompressed = this.storageProps.contains(HiveConstants.COMPRESSED)
+        ? Optional.of(this.storageProps.getPropAsBoolean(HiveConstants.COMPRESSED)) : Optional.<Boolean> absent();
+    this.numBuckets = this.storageProps.contains(HiveConstants.NUM_BUCKETS)
+        ? Optional.of(this.storageProps.getPropAsInt(HiveConstants.NUM_BUCKETS)) : Optional.<Integer> absent();
+    this.bucketColumns = this.storageProps.contains(HiveConstants.BUCKET_COLUMNS)
+        ? Optional.of(this.storageProps.getPropAsList(HiveConstants.BUCKET_COLUMNS)) : Optional.<List<String>> absent();
+    this.isSotredAsSubDirs = this.storageProps.contains(HiveConstants.STORED_AS_SUB_DIRS)
+        ? Optional.of(this.storageProps.getPropAsBoolean(HiveConstants.STORED_AS_SUB_DIRS))
+        : Optional.<Boolean> absent();
+
+    this.serDeType = Optional.fromNullable(this.serDeProps.getProp(HiveConstants.SERDE_TYPE));
+  }
+
+  public void setColumns(List<Column> columns) {
+    this.columns.clear();
+    this.columns.addAll(columns);
+  }
+
+  public void setProp(String key, Object value) {
```

Comment:
Can you add documentation as to the difference between `setProp` `setStorageProp` and `setSerDeProp`